airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/DatabaseConfigPersistence.java

```diff
@@ -1601,18 +1601,15 @@ public void replaceAllConfigs(final Map<AirbyteConfig, Stream<?>> configs, final
           sourceConnectionWithMetadata
               .stream()
               .map(configWithMetadata -> {
-                if (featureFlags.exposeSecretsInExport()) {
-                  return Jsons.jsonNode(configWithMetadata.getConfig());
-                }
-
                 try {
                   final UUID sourceDefinitionId = configWithMetadata.getConfig().getSourceDefinitionId();
                   final StandardSourceDefinition standardSourceDefinition = getConfig(
                       ConfigSchema.STANDARD_SOURCE_DEFINITION,
                       sourceDefinitionId.toString(),
                       StandardSourceDefinition.class);
                   final JsonNode connectionSpecs = standardSourceDefinition.getSpec().getConnectionSpecification();
-                  final JsonNode sanitizedConfig = jsonSecretsProcessor.maskSecrets(Jsons.jsonNode(configWithMetadata.getConfig()), connectionSpecs);
+                  final JsonNode sanitizedConfig = jsonSecretsProcessor.transformJson(Jsons.jsonNode(configWithMetadata.getConfig()),
```

Comment:
Not what I had in mind. You still have complex logic.

```
when you instantiate the DatabaseConfigPersistence:
JsonSecretsProcessor jsp = flag ? new RealJsonSecretsProcessor() : new NoOpJsonSecretsProcessor();

new DatabaseConfigPersistence(jsp);
```

This way you don't create more complex logic in the config persistence, the testing of that class is simple, you don't need to pass a colloection for feature flags.

Reply:
I changed it to a class, I am not 100% satisfy with it because we don't really want a no op for the copy secrets step since it is not affected by the previous comments. I think that I should change it to a factory instead to generate the JsonSecretsProcessor that will do or not do an operation.

Reply:
The current state is definitely in the vein of what I had in mind.

Reply:
I changed it to a factory. I prefer this implementation because while it is allowing to have a single pace for implementation, it allows to have several type of combination, like copy but don't mask, mask and copy, etc ...